modules/statistics/PrecallTest.js

```diff
@@ -0,0 +1,107 @@
+import browser from '../browser';
+import Settings from '../settings/Settings';
+import ScriptUtil from '../util/ScriptUtil';
+
+const SCRIPT_URL = 'https://api.callstats.io/static/callstats-ws.min.js';
+
+let _initialized = false;
+let api = null;
+
+/**
+ * Loads the callstats io script.
+ *
+ * @returns {Promise<void>}
+ */
+function _loadScript() {
+    if (browser.isReactNative()) {
+        return;
+    }
+
+    return new Promise(resolve => {
+        ScriptUtil.loadScript(
+            SCRIPT_URL,
+            /* async */ true,
+            /* prepend */ true,
+            /* relativeURL */ undefined,
+            /* loadCallback */ resolve);
+    });
+}
+
+/**
+ * Initializes the callstats lib and registers a callback to be invoked
+ * when there are 'preCallTestResults'.
+ *
+ * @typedef {Object} PrecallTestOptions
+ * @property {string} callStatsID - Callstats credentials - the id.
+ * @property {string} callStatsSecret - Callstats credentials - the secret.
+ * @property {string} statisticsId - The user name to use when initializing callstats.
+ * @property {string} statisticsDisplayName - The user display name.
+ *
+ * @param { PrecallTestOptions} options - The init options.
+ * @param {Function} onResult - The callback to be invoked when results are received.
+ * @returns {Promise<void>}
+ */
+function _initialize(options, onResult) {
+    return new Promise((resolve, reject) => {
+        if (!options.disableThirdPartyRequests) {
+            const appId = options.callStatsID;
+            const appSecret = options.callStatsSecret;
+            const userId = options.statisticsId || options.statisticsDisplayName || Settings.callStatsUserName;
+
+            api.initialize(appId, appSecret, userId, (status, message) => {
+                if (status === 'success') {
+                    api.on('preCallTestResults', (testStatus, result) => {
+                        if (testStatus === 'success') {
+                            onResult(result);
+                        }
+                    });
+                    _initialized = true;
+                    resolve();
+                } else {
+                    reject({
+                        status,
+                        message
+                    });
+                }
+            });
+        }
+    });
+}
+
+
+/**
+ * Loads the callstats script and initializes the library.
+ *
+ * @param {PrecallTestOptions} options - The init options.
+ * @param {Function} onResult - The callback to be invoked when results are received.
+ * @returns {Promise<void>}
+ */
+export async function init(options, onResult) {
+    if (_initialized) {
+        throw new Error('Precall Test already initialized');
+    }
+
+    await _loadScript();
+    // eslint-disable-next-line new-cap
+    api = new window.callstats();
```

Comment:
Please make sure this also works on RN. We use a module for that. We might need confirmation that the precall API is available there.

Reply:
Sorry, I don't know what *RN* is and how we should use the _precall test_ for it... Could you please give me some more details?

Reply:
Yes, tested it and this should work on React Native too.